### PR TITLE
[multistage] Retain timestampAdd and timestampDiff functions during SqlNode to RexNode conversion

### PR DESCRIPTION
**Pull request by xiangfu0**
### Background

CalciteSqlParser uses StandardConvertletTable to do series conversions for `SqlCall` -> `RexNode`.
One optimization it does is to convert `TimestampAdd` and `TimestampDiff` method to arithmetic timestamp and interval expression.
The problem we are facing here is that the literal expression of Interval is not supported which means the query is optimized to the form of Pinot doesn't recognize.
Hence the PR to not perform certain conversion in `org.apache.calcite.sql2rel.StandardConvertletTable`

### Changes
   - Add `PinotConvertletTable` which is a wrapper of `StandardConvertletTable` just override certain `SqlRexConvertlet`
   - Implements Pinot `TimestampAddConvertlet` and `TimestampDiffConvertlet` to keep the SqlCall as a function call format.
   - Refine `TimestampTests` to include more columns and coverage for `day`, `week`, `month`, `quarter`, `year` coverage.
   - Remove all `INTERVAL_*` types support in `RelToPlanNodeConverter` to avoid the confusion in v2

### CODE REVIEW

File: pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ */
+public class PinotConvertletTable implements SqlRexConvertletTable {
+
+  public static final PinotConvertletTable INSTANCE = new PinotConvertletTable();
+
+  private PinotConvertletTable() {
+  }
+
+  @Override
+  public @Nullable SqlRexConvertlet get(SqlCall call) {
```
Comment (Jackie-Jiang) [inline pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java:40]: (minor)
```suggestion
  @Nullable
  @Override
  public SqlRexConvertlet get(SqlCall call) {
```

File: pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    switch (call.getKind()) {
+      case TIMESTAMP_ADD:
+        return TimestampAddConvertlet.INSTANCE;
+      case TIMESTAMP_DIFF:
+        return TimestampDiffConvertlet.INSTANCE;
```
Comment (Jackie-Jiang) [inline pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java:45]: Does it work if we simply do
```suggestion
      case TIMESTAMP_ADD:
      case TIMESTAMP_DIFF:
        return null;
```

**Review by Jackie-Jiang — COMMENTED** [review]
Out of scope of this PR, but we should go over all the standard convertlet and only keep the ones needed. Keeping all of them can introduce too many surprises

File: pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java (hunk 2 above)
Comment (xiangfu0) [reply pinot-query-planner/src/main/java/org/apache/calcite/sql2rel/PinotConvertletTable.java:46]: Then there is no way to convert SqlNode to RexNode.
There has to be an implementation for every sqlkind.

**Comment by xiangfu0** [comment]
Correct, will do that

**Review by Jackie-Jiang — APPROVED** [review]

**Review by gortiz — APPROVED** [review]
Nice and elegant solution

**Pull request merged**